src/components/UserProfilePuzzle.tsx

```diff
@@ -1,9 +1,45 @@
 import '../css/UserProfilePuzzle.css';
+import Modal from 'react-modal';
+import PuzzleDetails from './PuzzleDetails'
+import { useState } from 'react'
+
+
+Modal.setAppElement('#root');
+
+interface UserPuzzleImage {
+  image: string
+  category: string
+  missingPieces: string 
+  price: string 
+  pieceCount: string 
+  quality: string
+  id: number | string 
+}
+
+const UserProfilePuzzle: React.FC<UserPuzzleImage> = ({ image, category, missingPieces, price, pieceCount, quality, id }) => {
+
+  const [modalOpen, setModalOpen] = useState(false)
+
+  const openModal = (event: React.MouseEvent) => {
```

Comment:
Super trivial ultimately, but it's usually better to destructure `MouseEvent` in your imports from React just like you do with useState. And then this can become 
```
const openModal = (e: MouseEvent) =>
```